crates/derive/src/sources/ethereum.rs

```diff
@@ -19,9 +19,9 @@ where
     B: BlobProvider + Clone,
 {
     /// The chain provider to use for the factory.
-    pub chain_provider: C,
+    pub chain_provider: Option<C>,
     /// The blob provider
-    pub blob_provider: B,
+    pub blob_provider: Option<B>,
```

Comment:
Hmm - all Ethereum providers should need both a chain provider and a blob source. Don't think we should make these optional?

Reply:
What if someone wants to configure a chain to have only one or the other?

Reply:
Good q - I think that this diverges from Ethereum DA-configured OP Stack chains, and should probably require a new data source implementation (which thankfully isn't hard now! Should def document this). The OP Stack specification describes a clear transition between calldata-available batches and blob-available batches with the transition from the Delta -> Ecotone hardfork, so I think the default source we provide should meet that spec.

Reply:
Reflecting that this is an OP Stack Ethereum DA in the type name might make this more clear. When I said 
>all Ethereum providers

I think I more meant "all OP Stack Ethereum DA providers"

Reply:
That's fine with me - you're right they can just define a new non-ethereum data source. Though the "Ethereum" data source name is then a misnomer - it should be `OPStackDataSource` ;P